framework/auth/core.py

```diff
@@ -1376,13 +1380,22 @@ def is_affiliated_with_institution(self, inst):
         return inst in self.affiliated_institutions
 
     def remove_institution(self, inst_id):
-        try:
-            self.affiliated_institutions.remove(inst_id)
-        except ValueError:
-            return False
-        return True
+        node_id = False
+        for node in self._affiliated_institutions:
+            if node.institution_id == inst_id:
+                node_id = node._id
+                break
+        if node_id:
+            self._affiliated_institutions.remove(node_id)
```

Comment:
`affiliated_institutions.remove()` instead of `_affiliated_institutions.remove()` here.